diff --git a/oauthproxy.go b/oauthproxy.go
index 17767ad..e2c524c 100644
--- a/oauthproxy.go
+++ b/oauthproxy.go
@@ -546,7 +546,6 @@ func (p *OAuthProxy) ErrorPage(rw http.ResponseWriter, code int, title string, m
 // SignInPage writes the sing in template to the response
 func (p *OAuthProxy) SignInPage(rw http.ResponseWriter, req *http.Request, code int) {
 	p.ClearSessionCookie(rw, req)
-	rw.WriteHeader(code)

 	redirecURL := req.URL.RequestURI()
 	if req.Header.Get("X-Auth-Request-Redirect") != "" {
@@ -556,6 +555,14 @@ func (p *OAuthProxy) SignInPage(rw http.ResponseWriter, req *http.Request, code
 		redirecURL = "/"
 	}

+	if p.SkipProviderButton {
+		req.Form.Set("rd", redirecURL)
+		p.OAuthStart(rw, req)
+		return
+	}
+
+	rw.WriteHeader(code)
+
 	t := struct {
 		ProviderName  string
 		SignInMessage string
@@ -696,11 +704,7 @@ func (p *OAuthProxy) SignIn(rw http.ResponseWriter, req *http.Request) {
 		p.SaveSession(rw, req, session)
 		http.Redirect(rw, req, redirect, 302)
 	} else {
-		if p.SkipProviderButton {
-			p.OAuthStart(rw, req)
-		} else {
-			p.SignInPage(rw, req, http.StatusOK)
-		}
+		p.SignInPage(rw, req, http.StatusOK)
 	}
 }
